cli/azd/internal/cmd/add/add_configure_host.go

```diff
@@ -231,50 +264,70 @@ func addServiceAsResource(
 	ctx context.Context,
 	console input.Console,
 	svc *project.ServiceConfig,
-	prj appdetect.Project) (*project.ResourceConfig, error) {
+	prj appdetect.Project,
+) (*project.ResourceConfig, error) {
 	resSpec := project.ResourceConfig{
 		Name: svc.Name,
 	}
-
-	if svc.Host == project.ContainerAppTarget {
-		resSpec.Type = project.ResourceTypeHostContainerApp
-	} else {
+	if svc.Host != project.ContainerAppTarget && svc.Host != project.AppServiceTarget {
 		return nil, fmt.Errorf("unsupported service target: %s", svc.Host)
 	}
 
-	props := project.ContainerAppProps{
-		Port: -1,
-	}
+	port := -1
 
 	if svc.Docker.Path == "" {
 		// no Dockerfile is present, set port based on azd default builder logic
 		if _, err := os.Stat(filepath.Join(svc.RelativePath, "Dockerfile")); errors.Is(err, os.ErrNotExist) {
 			// default builder always specifies port 80
-			props.Port = 80
+			port = 80
 			if svc.Language == project.ServiceLanguageJava || svc.Language.IsDotNet() {
-				props.Port = 8080
+				port = 8080
 			}
 		}
 	}
```

Comment:
@weikanglim @puicchan any thoughts on whether we should prompt for the port number in app service case? Right now it just defaults to 80, but can be changed in `azure.yaml`

Reply:
Defaulting and exposing it is fine.